lib/kill.js

```diff
@@ -65,6 +77,9 @@ export const validateTimeout = ({timeout}) => {
 	}
 };
 
+// Used only to interrupt control flow, but not used in return value
+export const TEMPORARY_ERROR = new Error('Temporary');
```

Comment:
For timeouts, non-0 exit codes and signals, we do not want to add anything to the error message. We are also not interested in the stack trace, since the error would be created inside Execa code base. 

Instead, we rely on the main `makeError()` logic to set the right error message, and create some stack trace.